internal/bgp/frr/frr_test.go

```diff
@@ -51,18 +53,26 @@ func testGenerateFileNames(t *testing.T) (string, string) {
 func testSetup(t *testing.T) {
 	configFile, _ := testGenerateFileNames(t)
 	os.Setenv("FRR_CONFIG_FILE", configFile)
-	_, err := os.Create(configFile)
-	if err != nil {
-		t.Fatalf("cannot create config file %s: %s", configFile, err)
-	}
-
+	_ = os.Remove(configFile) // removing leftovers from previous runs
 	osHostname = testOsHostname
 }
 
 func testCheckConfigFile(t *testing.T) {
-	time.Sleep(10 * time.Millisecond) // give time to the debouncer to write the file
 	configFile, goldenFile := testGenerateFileNames(t)
+	err := wait.PollImmediate(10*time.Millisecond, 2*time.Second, func() (bool, error) {
+		_, err := os.Stat(configFile)
+		if errors.Is(err, os.ErrNotExist) {
+			return false, nil
+		}
+		if err != nil {
+			return false, err
+		}
+		return true, nil
+	})
 
+	if err != nil {
+		t.Fatalf("Failed to wait the file %s", configFile)
```

Comment:
maybe "Failed to wait for configfile %s err %s" ?

Reply:
yes, good point. Changing it.